src/sentry/utils/event_frames.py

```diff
@@ -106,6 +119,23 @@ def get_sdk_name(event_data: PathSearchable) -> Optional[str]:
     return get_path(event_data, "sdk", "name", filter=True) or None
 
 
+def try_munge_frame_path(
```

Comment:
This is a new function that will replace the below function `munged_filename_and_frames()`

The old function accepted multiple frames at once and mutated the frame object to add the munged filename as one of the keys. This was bad for typing and unnecessarily complicated IMO. The new function simply returns the munged filename or `None`.

Reply:
The places that use this are:

- Suspect commits (will convert in next PR)
- Stacktrace link (will convert in next PR)
- Committers endpoint (this should be a simple conversion)
- Ownership rules (this one might be more difficult)

I'm leaving the current function untouched for now so it will continue to work until we replace the others. I think it's okay to have some duplicated code for a bit.